subsys/bluetooth/host/Kconfig

```diff
@@ -991,36 +991,21 @@ config BT_DF_VS_CONN_IQ_REPORT_16_BITS_IQ_SAMPLES
 	  format.
 
 endif # BT_DF
-endif # BT_HCI_HOST
 
 config BT_ECC
-	bool "ECDH key generation support"
-	default y if BT_SMP && !BT_SMP_OOB_LEGACY_PAIR_ONLY
-	help
-	  This option adds support for ECDH HCI commands.
-
-config BT_SEND_ECC_EMULATION
-	bool "Emulate ECDH in the Host using PSA Crypto API library"
+	bool
 	select MBEDTLS if !BUILD_WITH_TFM
 	select MBEDTLS_PSA_CRYPTO_C if !BUILD_WITH_TFM
 	select PSA_WANT_ALG_ECDH
 	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_GENERATE
 	select PSA_WANT_ECC_SECP_R1_256
 	imply MBEDTLS_PSA_P256M_DRIVER_ENABLED if MBEDTLS_PSA_CRYPTO_C
 	select BT_LONG_WQ
-	depends on BT_ECC && (BT_HCI_RAW || BT_HCI_HOST)
-	default y if HAS_BT_CTLR && !BT_CTLR_ECDH
-	help
-	  If this option is set PSA Crypto API library is used for emulating the
-	  ECDH HCI commands and events needed by e.g. LE Secure Connections.
-	  In builds including the BLE Host, if not set the controller crypto is
-	  used for ECDH and if the controller doesn't support the required HCI
-	  commands the LE Secure Connections support will be disabled.
-	  In builds including the HCI Raw interface and the BLE Controller, this
-	  option injects support for the 2 HCI commands required for LE Secure
-	  Connections so that Hosts can make use of those. The option defaults
-	  to enabled for a combined build with Zephyr's own controller, since it
-	  does not have any special ECC support itself (at least not currently).
+	help
+	  If this option is set, internal APIs will be available to perform ECDH operations
+	  through the long work queue. operations needed e.g. by LE Secure Connections.
```

Comment:
Is my understanding correct that even if the controller supports the ECDH HCI commands, then we now always include it in the host? 

I thought we had this split for the cases where the controller supports ECDH over HCI, that it would needed to support in the host as well. 

In dual-CPU devices like the nRF5340 then we'll now always have a ECDH implemention in the host, thus increasing the resource requirements, right?

Reply:
> Is my understanding correct that even if the controller supports the ECDH HCI commands, then we now always include it in the host?

Correct. However, this was in most cases purely theoretical, i.e. we never really had such controllers in practice.

> In dual-CPU devices like the nRF5340 then we'll now always have a ECDH implemention in the host, thus increasing the resource requirements, right?

Was that really the case? If so, then I'd have missed it. I thought even these platforms did the operations in the app core.

Reply:
I can't at least find any ECDH support when greping around the upstream controller code, so if there ever was the opportunity to use controller side ECDH HCI commands & events it wasn't with that controller.

Reply:
> I can't at least find any ECDH support when greping around the upstream controller code, so if there ever was the opportunity to use controller side ECDH HCI commands & events it wasn't with that controller.

Actually wait. Our hci_raw implementation was able to insert the same emulation layer into the HCI stream, so the controller below didn't need to support those commands. Anyway, it would be valuable to know if this is really a strong requirement for Nordic or not, since otherwise I'd be in favor of simplifying all of this and removing unnecessary overhead (just look at the diffstat of this PR).

Reply:
In ncs we implement these commands outside the controller itself (https://github.com/nrfconnect/sdk-nrf/blob/1e301e40ae1d342703b1226979dce15e8690716c/subsys/bluetooth/controller/ecdh.c), so personally I would be fine with relying on ECDH in the host. @rugeGerritsen might have some input here.

I think this would only be a problem for someone using a non-zephyr host. If both the controller and host side are zephyr then developer can now disable ECDH in the controller and enable it in the host instead. This should be more efficient since they no longer have to go over HCI.